dra-evolution/pkg/api/capacity_types.go

```diff
@@ -78,37 +75,70 @@ type ResourcePoolSpec struct {
 	// them) empty pool.
 }
 
-const ResourcePoolMaxSharedCapacity = 128
+const ResourcePoolMaxCommonData = 16
 const ResourcePoolMaxDevices = 128
 
+type CommonDeviceData struct {
+	// Name identifies the particular set of common data represented
+	// here.
+	//
+	// +required
+	Name string `json:"name"`
+
+	// Exactly one must be populated
+	Attributes []DeviceAttribute
+	Capacity   []DeviceCapacity
+
+	// Future possible additions:
+	// - SharedCapacity for use in allocating partitionable devices
+	// - DeviceShape for encapsulating attributes, capacities, partion schemes
+	// - DeviceShapeRef for storing those in a secondary object
+}
+
 // Device represents one individual hardware instance that can be selected based
 // on its attributes.
 type Device struct {
 	// Name is unique identifier among all devices managed by
 	// the driver on the node. It must be a DNS label.
 	Name string `json:"name" protobuf:"bytes,1,name=name"`
 
+	// CommonData is the list of names of entries from the common data that
+	// should be included in this device.
+	//
+	// +listType=atomic
+	// +optional
+	//
+	CommonData []string `json:"commonData,omitempty"`
```

Comment:
As as mentioned above -- I would remove this until we know how we are going to support partitioning.